include/circt/Dialect/Arc/ArcOps.td

```diff
@@ -905,4 +952,74 @@ def VectorizeReturnOp : ArcOp<"vectorize.return", [
   let assemblyFormat = "operands attr-dict `:` qualified(type(operands))";
 }
 
+def EnvironmentCallOp : ArcOp<"environment_call", [
```

Comment:
For now there is no difference. But when toying around with the runtime library implementation I encountered several things that could require separating environment calls from other functions:
- It could be helpful to export a list of the environment calls a specific model needs to the JSON file. Potentially just as an indicator that the model _does_ require the runtime library.
- Having them separate gives us more control over the eventual lowering to LLVM IR functions and calls. This could come in handy to specify the linkage or deal with calls that are 'exceptional', e.g., terminating simulation.
- At some point I probably want to do type marshaling, i.e, allow environment calls with "arbitrary" IR types and potentially multiple return values. This will require a dedicated transformation pass before LLVM lowering.

This is all purely hypothetical at the moment.  And I'll move the EnvironmentCallOp and CallEnvironmentOp to another PR before landing this one.